### PR TITLE
CLN, TYP: Factor out part of info

### PR DESCRIPTION
**Pull request by MarcoGorelli**
Precursor to #31796, which still has a rather large diff. This should help make it easier to read.

On master, `cols` holds the DataFrame's columns. If we were to use this method on a Series, then it would hold an Index containing the Series' name (`Index([data.name])`). Is there a good name for a variable that could store either? At the moment I've called it `ids`

### CODE REVIEW

File: pandas/io/formats/info.py [hunk 1]
```diff
@@ -1,18 +1,28 @@
...
+    from pandas.core.indexes.api import Index  # noqa: F401
+    from pandas.core.series import Series  # noqa: F401
 
-def _put_str(s, space):
+
+def _put_str(s: Union[str, Dtype], space: int) -> str:
     return str(s)[:space].ljust(space)
 
 
+def _get_ids_and_dtypes(data: FrameOrSeries) -> Tuple["Index", "Series"]:
```
Comment (jreback) [inline pandas/io/formats/info.py:20]: where is this useful?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/io/formats/info.py (hunk 1 above)
Comment (MarcoGorelli) [reply pandas/io/formats/info.py:20]: @jreback if I rebase #31796 onto this, the diff is easier to read as all that needs to change for this part is a
```python
if isinstance(data, Series)
```
inside this helper function

File: pandas/io/formats/info.py [hunk 2]
```diff
@@ -1,18 +1,28 @@
...
 from pandas.io.formats import format as fmt
 from pandas.io.formats.printing import pprint_thing
 
+if TYPE_CHECKING:
+    from pandas.core.indexes.api import Index  # noqa: F401
+    from pandas.core.series import Series  # noqa: F401
 
-def _put_str(s, space):
+
+def _put_str(s: Union[str, Dtype], space: int) -> str:
```
Comment (jreback) [inline pandas/io/formats/info.py:16]: can you add a doc-string

File: pandas/io/formats/info.py (hunk 1 above)
Comment (jreback) [reply pandas/io/formats/info.py:20]: ok, please add a doc-string

**Review by jreback — CHANGES_REQUESTED** [review]
@MarcoGorelli ok, some comments, ping on green.

**Comment by MarcoGorelli** [comment]
>  ping on green.

@jreback ping

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @MarcoGorelli

**Pull request merged**